core/src/main/java/org/elasticsearch/rest/MethodHandlers.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Encapsulate multiple handlers for the same path, allowing different handlers for different HTTP verbs.
+ */
+public class MethodHandlers {
```

Comment:
> I think, @dakrone correct me if I am wrong, that in #24437 we need to know if a rest handler can handle a certain verb to ensure we send the right error back. That might be the reason why we need to be explicit about it. But maybe there is a way to do this?

That's correct, that's the reason for this, the only way I can think of to do it on the handler itself is collapsing every handler for the same rest endpoint into the same class, and then duplicating methods to determine which verbs are supported. I don't think it would be cleaner at all